### PR TITLE
chore(consensus): Move get_block_maker_by_rank into test utilities

### PR DESCRIPTION
**Pull request by dist1ll**
I plan to use this function from inside the `test_utilities` crate. Beyond that, this change removes a good chunk of code.

### CODE REVIEW

File: rs/test_utilities/artifact_pool/src/consensus_pool.rs [hunk 1]
```diff
@@ -196,15 +197,35 @@ impl TestConsensusPool {
...
+    // specified rank at a given height. Panics if this rank does not exist.
+    pub fn get_block_maker_by_rank(&self, height: Height, rank: Rank) -> NodeId {
+        let pool_reader = PoolReader::new(&self.pool);
+        *self
+            .membership
+            .get_nodes(height)
+            .unwrap()
+            .iter()
+            .find(|node| {
+                let prev_beacon = pool_reader.get_random_beacon(height.decrement()).unwrap();
```
Comment (kpop-dfinity) [inline rs/test_utilities/artifact_pool/src/consensus_pool.rs:221]: nit: can we move this line out of the closure?

File: rs/test_utilities/artifact_pool/src/consensus_pool.rs [hunk 2]
```diff
@@ -196,15 +197,35 @@ impl TestConsensusPool {
...
             registry_client,
             pool,
             time_source,
             dkg_payload_builder,
+            membership,
         }
     }
 
+    // Utility function to determine the identity of the block maker with the
+    // specified rank at a given height. Panics if this rank does not exist.
```
Comment (kpop-dfinity) [inline rs/test_utilities/artifact_pool/src/consensus_pool.rs:212]: nit:
```suggestion
    /// Utility function to determine the identity of the block maker with the
    /// specified rank at a given height. Panics if this rank does not exist.
```

**Review by kpop-dfinity — APPROVED** [review]

File: rs/test_utilities/artifact_pool/src/consensus_pool.rs (hunk 1 above)
Comment (dist1ll) [reply rs/test_utilities/artifact_pool/src/consensus_pool.rs]: Done

File: rs/test_utilities/artifact_pool/src/consensus_pool.rs (hunk 2 above)
Comment (dist1ll) [reply rs/test_utilities/artifact_pool/src/consensus_pool.rs]: Changed, thanks

**Pull request merged**